lib/plugins/aws/info/getStackInfo.js

```diff
@@ -52,6 +43,11 @@ module.exports = {
           .then(result => {
             if (result) stackData.externalHttpApiEndpoint = result.ApiEndpoint;
           })
+          .catch(e => {
```

Comment:
This should be a _second_ callback to previous `then` (not a new `catch` clause), for reasons outlined here: https://github.com/serverless/serverless/pull/8157#discussion_r481039055

Eventually we can make it as `catch` clause that directly follows `provider.request` above

Reply:
understood your point, updated.

We cannot make it a catch clause in resolveCfImportValue as Serverless.Error class is not accessible there

Reply:
However I ran some test, and I think final catching in this use case is quite useful.

**With the code as is in this current PR state**, if I reference a non-existing name for my httpAPI id import, I get the following error :

![image](https://user-images.githubusercontent.com/23530626/91841184-30e4dd00-ec52-11ea-8325-75894b2b04d6.png)

And when I reference an existing import name, but the corresponding value does not match any httpApi Id, I get the following error :

![image](https://user-images.githubusercontent.com/23530626/91841299-64c00280-ec52-11ea-97b1-f1e8f3d96d9c.png)

If I change this to be catching only error from `provider.request` using `then` second argument or a `catch` clause on `request` method, the first error remain the same, will the second become much less clear as to where and why the error occured, since I'm only left with the `Invalid API identifier specified 871803655707:value` error message.

Don't you think the current version is actually better for DX ?

Reply:
In case I commented, the only thing I proposed, is to not cover this callback:

```javascript
if (result) stackData.externalHttpApiEndpoint = result.ApiEndpoint;
```

with added error handler.  I don't see how it can produce difference you point in screenshots. By content of callback it seems clear that it's only programmer error that can eventually let surface, not AWS request error

Reply:
To be perfectly clear. I mean to do this:

```javascript
.then(id => {
  return this.provider.request('ApiGatewayV2', 'getApi', { ApiId: id });
})
.then(result => {
  stackData.externalHttpApiEndpoint = result.ApiEndpoint;
}, e => {
  throw new this.serverless.classes.Error(
    `Could not resolve provider.httpApi.id parameter. ${e.message}`
  );
})
```

instead of

```javascript
.then(id => {
  return this.provider.request('ApiGatewayV2', 'getApi', { ApiId: id });
})
.then(result => {
  stackData.externalHttpApiEndpoint = result.ApiEndpoint;
})
.catch(e => {
  throw new this.serverless.classes.Error(
    `Could not resolve provider.httpApi.id parameter. ${e.message}`
  );
})
```

Reply:
Ok, understood your point about leaving out the last `then` statement.

If i proceed and add a second argument as below :

```js
(httpApiId['Fn::ImportValue']
  ? resolveCfImportValue(this.provider, httpApiId['Fn::ImportValue'])
  : BbPromise.resolve(httpApiId)
)
  .then(
    id => {
      return this.provider.request('ApiGatewayV2', 'getApi', { ApiId: id });
    },
    e => {
      throw new this.serverless.classes.Error(
        `Could not resolve provider.httpApi.id parameter. ${e.message}`
      );
    }
  )
```

Any error thrown by `this.provider.request('ApiGatewayV2', 'getApi', { ApiId: id })` will not be catched by `then` second argument - the `onRejected` callback, and the resulting error message will be as it is currently on master for the developer, meaning only the following indication : `Invalid API identifier specified 871803655707:value`

However, it is not the concern of this PR. I'll update according to your comments